src/main/java/ca/corefacility/bioinformatics/irida/model/enums/ProjectRole.java

```diff
@@ -1,39 +1,75 @@
-
 package ca.corefacility.bioinformatics.irida.model.enums;
 
+import java.util.Collection;
+
+import ca.corefacility.bioinformatics.irida.model.joins.impl.ProjectUserJoin;
+import ca.corefacility.bioinformatics.irida.model.user.group.UserGroupProjectJoin;
+
 /**
  *
  */
 public enum ProjectRole {
+	PROJECT_USER("PROJECT_USER", 1),
+	PROJECT_OWNER("PROJECT_OWNER", 2);
+
+	private String code;
+	private int level;
+
+	private ProjectRole(String code, int level) {
+		this.code = code;
+		this.level = level;
+	}
+
+	@Override
+	public String toString() {
+		return code;
+	}
+
+	/**
+	 * Get a role from the given string code
+	 *
+	 * @param code the string to get a role for
+	 * @return The requested ProjectRole
+	 */
+	public static ProjectRole fromString(String code) {
+		switch (code.toUpperCase()) {
+		case "PROJECT_USER":
+			return PROJECT_USER;
+		case "PROJECT_OWNER":
+			return PROJECT_OWNER;
+		default:
+			return PROJECT_USER;
+		}
+	}
+
+	public int getLevel() {
+		return level;
+	}
+
+	/**
+	 * Compares and returns the highest level {@link ProjectRole} from a {@link ProjectUserJoin} and a collection of
+	 * {@link UserGroupProjectJoin}s.
+	 *
+	 * @param projectUserJoin       a user's {@link ProjectUserJoin}
+	 * @param userGroupProjectJoins a collection of {@link UserGroupProjectJoin}
+	 * @return the max {@link ProjectRole}
+	 */
+	public static ProjectRole getMaxRoleForProjectsAndGroups(ProjectUserJoin projectUserJoin,
+			Collection<UserGroupProjectJoin> userGroupProjectJoins) {
+		ProjectRole projectRole = null;
+
+		if (projectUserJoin != null) {
+			projectRole = projectUserJoin.getProjectRole();
+		}
+
+		for (UserGroupProjectJoin userGroupProjectJoin : userGroupProjectJoins) {
+			if (projectRole == null || projectRole.getLevel() < userGroupProjectJoin.getProjectRole().getLevel()) {
```

Comment:
If a User does not have a projectUserJoin entry, but are members of multiple UserGroups and the first one has the role "PROJECT_USER" won't the loop end at this point and produce the incorrect output (Assuming they are a member of another UserGroup that has "PROJECT_OWNER" role. I think the only time that the loop should break is if we find a projectRole that equals "PROJECT_OWNER".

Reply:
Please see 6d8bf87762495667ad9c08695d5e678b3b4b1abb.